src/vllm_tgis_adapter/grpc/grpc_server.py

```diff
@@ -242,7 +243,7 @@ async def Generate(
             )
             is_tracing_enabled = await self.engine.is_tracing_enabled()
             headers = dict(context.invocation_metadata())
-            logs.set_correlation_id(request_id_i, headers.get("x-correlation_id"))
+            logs.set_correlation_id(request_id_i, headers.get(CORRELATION_ID_HEADER))
```

Comment:
:thinking:  `headers.get()` might return `None`, but `set_correlation_id` handles `str` only

Reply:
Yeah, mostly just a type hinting issue there but I updated to not touch the cache if there's no correlation id